lib/ansible/plugins/inventory/azure_rm.py

```diff
@@ -511,12 +527,38 @@ def hostvars(self):
                     new_hostvars['private_ipv4_addresses'].append(private_ip)
                 pip_id = ipc['properties'].get('publicIPAddress', {}).get('id')
                 if pip_id:
+                    new_hostvars['public_ip_id'] = pip_id
+
                     pip = nic.public_ips[pip_id]
+                    new_hostvars['public_ip_name'] = pip._pip_model['name']
                     new_hostvars['public_ipv4_addresses'].append(pip._pip_model['properties'].get('ipAddress', None))
                     pip_fqdn = pip._pip_model['properties'].get('dnsSettings', {}).get('fqdn')
                     if pip_fqdn:
                         new_hostvars['public_dns_hostnames'].append(pip_fqdn)
 
+            new_hostvars['mac_address'] = nic._nic_model['properties'].get('macAddress') if nic._nic_model['properties'].get('macAddress') else None
+            new_hostvars['network_interface'] = nic._nic_model['name']
+            new_hostvars['network_interface_id'] = nic._nic_model['id']
+
+        # set image and os_disk
+        new_hostvars['image'] = {}
+        new_hostvars['os_disk'] = {}
+        storageProfile = self._vm_model['properties'].get('storageProfile')
+        if storageProfile:
+            imageReference = storageProfile.get('imageReference')
+            new_hostvars['image'] = dict(
+                sku=imageReference.get('sku'),
```

Comment:
if i am correct, that image reference may contain 'id' instead of sku/publisher/version/offer

Reply:
just do sth like:
```
            if image.get('publisher'):
                new_result['image'] = {
                    'publisher': image['publisher'],
                    'sku': image['sku'],
                    'offer': image['offer'],
                    'version': image['version']
                }
            else:
                new_result['image'] = {
                    'id': image.get('id')
                }
```

Reply:
sample imageReference
```
imageReference': {
                u 'sku': u '16.04.0-LTS', u 'publisher': u 'Canonical', u 'version': u 'latest', u 'offer': u 'UbuntuServer'
            }
```